FAILED: PersistentActorFailureSpec

https://jenkins.akka.io:8498/job/akka-nightly/3602/consoleFull

Several failures, first is:
```
[info] - must call onRecoveryFailure when recovery from persisted events fails *** FAILED *** (9 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected class akka.actor.ActorRef, found class java.lang.String (recovery-failure:blahonga 1 1)
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:436)
[info]   at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:408)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:737)
[info]   at akka.persistence.PersistentActorFailureSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply$mcV$sp(PersistentActorFailureSpec.scala:185)
[info]   at akka.persistence.PersistentActorFailureSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(PersistentActorFailureSpec.scala:175)
[info]   at akka.persistence.PersistentActorFailureSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(PersistentActorFailureSpec.scala:175)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:57)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike
```

--- a/akka-persistence/src/test/scala/akka/persistence/PersistentActorFailureSpec.scala
+++ b/akka-persistence/src/test/scala/akka/persistence/PersistentActorFailureSpec.scala
@@ -6,11 +6,13 @@ package akka.persistence
 
 import akka.actor.{ OneForOneStrategy, _ }
 import akka.persistence.journal.AsyncWriteJournal
-import akka.testkit.{ EventFilter, ImplicitSender, TestEvent }
+import akka.testkit.{ EventFilter, ImplicitSender, TestEvent, TestProbe }
+
 import scala.collection.immutable
 import scala.util.control.NoStackTrace
 import scala.util.{ Failure, Try }
 import akka.persistence.journal.inmem.InmemJournal
+
 import scala.concurrent.Future
 
 object PersistentActorFailureSpec {
@@ -181,9 +183,13 @@ class PersistentActorFailureSpec extends PersistenceSpec(PersistenceSpec.config(
       expectMsg(List("corrupt"))
 
       // recover by creating another with same name
-      system.actorOf(Props(classOf[Supervisor], testActor)) ! props
+      // note that if we used testActor as failure detector passed in
+      // the props we'd have a race on our hands (#21229)
+      val failProbe = TestProbe()
+      val sameNameProps = Props(classOf[OnRecoveryFailurePersistentActor], name, failProbe.ref)
+      system.actorOf(Props(classOf[Supervisor], testActor)) ! sameNameProps
       val ref = expectMsgType[ActorRef]
-      expectMsg("recovery-failure:blahonga 1 1")
+      failProbe.expectMsg("recovery-failure:blahonga 1 1")
       watch(ref)
       expectTerminated(ref)
     }

